iOS Client/DevotionsController/MBDevotionsViewController.swift

```diff
@@ -49,12 +77,29 @@ class MBDevotionsViewController: UIViewController, StoreSubscriber, UITableViewD
         case .loading:
             break
         case .loaded(let loadedDevotions):
-            self.devotions = loadedDevotions
-            tableView.reloadData()
+            if self.devotions.count == 0 {
+                self.devotions = loadedDevotions
+                tableView.reloadData()
+            }
+        }
+    }
+    
+    func scrollToSelectedDevotion(animated: Bool) {
+        guard let selectedRow = devotions.index(where: { (devotion) -> Bool in
+            if let selectedDate = self.latestSelectedDate?.convertedDate(),
```

Comment:
How does `self.latestSelectedDate` get set for the first time? I see that when you select a date on the calendar it gets set, but what date does it scroll to when this gets called in `viewDidLoad`?

Reply:
So I call this from two places: `viewDidAppear` and `didSelectDayView`. As you mentioned, `didSelectDayView` sets the `latestSelectedDate`.

The trick is, the calendar calls `didSelectDayView` with the current date when it loads, so it's ready to go in `viewDidAppear`. I think I avoided doing stuff in `viewDidLoad` b/c it was too early to scroll the table view since the frames weren't all setup yet (the scroll was slightly off). The library made me put those two lines in `viewDidLayoutSubviews` and I think it resizes itself a bit there.